### PR TITLE
Alternate job titles

### PR DESCRIPTION
**Pull request by MarkSuckerberg**
## About The Pull Request
(Substantial amount of code ported from: Skyrat-SS13/Skyrat13#1887, but lots adjusted to fit the codebase)

Adds what it says in the title.
image
(Cargo tech and CE now have alternate titles, but I didn't want to recompile bc I'm lazy)

Also adds support for alternate job titles still showing the original ID and HUD icons:
image

**SEE `_alt_titles.dm` FOR DETAILS**

## Why It's Good For The Game
Adds a little bit more individuality(???) to the game without making it confusing by changing ID or HUD icons.

## Changelog
:cl:
add: Added alternate job titles to the game
tweak: If you set an ID's job to an alternate job title, it will actually show the proper overlay.
/:cl:

### CODE REVIEW

**Comment by SynnGraffkin** [comment]
Is this going to include job loadouts being minorly changed based on selected job? For example, a Doctor who chooses "Surgeon" as their title spawning with the Surgeon clothes on instead of Doctor outfit? Or a Doctor who selects "Nurse" spawning in nurse clothes?

**Comment by SynnGraffkin** [comment]
You can find Doctor, for example, here from VG. 

https://github.com/vgstation-coders/vgstation13/blob/47b0f49f319439469f51f99356a0bf1a8c4f05a9/code/game/jobs/job/medical.dm

**Comment by MarkSuckerberg** [comment]
I can see if that's possible, but I'm unsure if I'll be able to do it without just making the alternate titles jobs in themselves proper job datums, or... actually, let me try something

File: code/datums/datacore.dm [hunk 1]
```diff
@@ -1,314 +1,319 @@
...
-
-/datum/datacore/proc/get_id_photo(mob/living/carbon/human/H, client/C, show_directions = list(SOUTH))
-	var/datum/job/J = SSjob.GetJob(H.mind.assigned_role)
-	var/datum/preferences/P
-	if(!C)
-		C = H.client
-	if(C)
-		P = C.prefs
-	return get_flat_human_icon(null, J, P, DUMMY_HUMAN_SLOT_MANIFEST, show_directions)
+//TODO: someone please get rid of this shit
```
Comment (Maldaris) [inline code/datums/datacore.dm:1]: get rid of it marg

Comment (MarkSuckerberg) [reply code/datums/datacore.dm:1]: holdup let me delet

**Comment by dee-dubya** [comment]
we config yet?

**Comment by Superyodeler** [comment]
Looks like all custom job names are put into the Misc category on the crew manifest

File: waspstation/code/modules/jobs/job_types/_alt_titles.dm [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+/datum/job/mime
+	alt_titles = list("Performer")
+
+//Science
+/datum/job/rd
+	alt_titles = list("Research Supervisor", "Head Scientist", "Chief Researcher", "Chief Science Officer")
+
+/datum/job/scientist
+	alt_titles = list("Xenobiologist", "Chemical Researcher", "Toxologist", "Nanite Researcher", "Junior Scientist")
```
Comment (Superyodeler) [inline waspstation/code/modules/jobs/job_types/_alt_titles.dm:60]: I feel like Chemical Researcher and Toxologist are too easily mistaken for chemist. In fact, Chemical Researcher might be a good for chem to have.
```suggestion
	alt_titles = list("Xenobiologist", "Nanite Researcher", "Junior Scientist")
```

File: waspstation/code/modules/jobs/job_types/_alt_titles.dm [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+/datum/job/paramedic
+	alt_titles = list("Emergency Medical Technician")
+
+/datum/job/chemist
+	alt_titles = list("Pharmacist", "Pharmacologist", "Junior Chemist")
+	senior_title = "Senior Chemist"
+
+/datum/job/geneticist
+	alt_titles = list("Biochemist")
```
Comment (Superyodeler) [inline waspstation/code/modules/jobs/job_types/_alt_titles.dm:83]: I think this one might be confusing too

File: waspstation/code/modules/jobs/job_types/_alt_titles.dm [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+/datum/job/detective
+	alt_titles = list("Forensics Technician", "Private Investigator")
+	senior_title = "Hard-boiled Detective"
+
+/datum/job/brig_phys
+	alt_titles = list("Security Medic")
+
+/datum/job/officer
+	alt_titles = list("Security Guard", "Junior Officer", "Prison Guard")
```
Comment (Superyodeler) [inline waspstation/code/modules/jobs/job_types/_alt_titles.dm:103]: Not sure about Prison Guard, with the low number of seccies we have around it's probably not a good idea to have them hanging around brig all shift. That's kinda the warden's job.

**Comment by dinnerwoerror** [comment]
yea literally just get rid of the stupid titles, do we really need four different names for the same job

**Comment by Victor239** [comment]
Agreed, no snowflake titles

**Comment by Superyodeler** [comment]
Some people were unhappy when they got untestmerged. I'm kinda unsure about them, but if you can clearly tell what job they're referring to and they show up correctly on huds etc I don't see any major problems.

**Comment by Maldaris** [comment]
> Some people were unhappy when they got untestmerged. I'm kinda unsure about them, but if you can clearly tell what job they're referring to and they show up correctly on huds etc I don't see any major problems.

So there was a reported issue with alternate titles showing up in the Misc category when selected players are listed (I think in the manifest?)

Also this is ammo for me insisting we get rid of the #bug-report channel we lose more info than we gain with that channel.

**Comment by Victor239** [comment]
Or we get a bot for that channel so you can create issues directly from it
But seeing as how unreliable even kiwi is this is not likely

**Comment by Maldaris** [comment]
> Or we get a bot for that channel so you can create issues directly from it
> But seeing as how unreliable even kiwi is this is not likely

Kiwi's a special case and we're getting better about it. With a third party bot we might be able to get this.

**Comment by Superyodeler** [comment]
I think the alt titles showing up under misc has been fixed now.

**Comment by MarkSuckerberg** [comment]
as much as I don't want to ask it, like I said previously, short of someone making the complete list for me, I'm not touching the titles

File: waspstation/code/modules/jobs/job_types/_alt_titles.dm (hunk 3 above)
Comment (Maldaris) [reply waspstation/code/modules/jobs/job_types/_alt_titles.dm]: Removed

File: waspstation/code/modules/jobs/job_types/_alt_titles.dm (hunk 2 above)
Comment (Maldaris) [reply waspstation/code/modules/jobs/job_types/_alt_titles.dm]: Removed

**Comment by Maldaris** [comment]
Changes per review have been made, as well as margularization comments. This should be good to go @Superyodeler

**Comment by SynnGraffkin** [comment]
> as much as I don't want to ask it, like I said previously, short of someone making the complete list for me, I'm not touching the titles

I've posted the entire title list from VG before.

**Comment by dee-dubya** [comment]
regarding the list, wasn't the plan to make this a config? or is that no longer happening

**Comment by Superyodeler** [comment]
Sounds like it's not happening

**Comment by SynnGraffkin** [comment]
I don't remember alt titles ever having a discussion for config settings.

**Comment by Maldaris** [comment]
> regarding the list, wasn't the plan to make this a config? or is that no longer happening

There are runtime things like alt-title-based outfits that don't jive with configs. See [this](https://discordapp.com/channels/659954234212745235/660934794250682388/725113891369123850) conversation in discord for an explanation. If we were to alter the title ingame and there was an associated outfit it would disappear.

**Comment by TheNeoGamer42** [comment]
this was discussed in the discord, but there shouldn't be alt titles for heads, it only adds for confusion

**Comment by Maldaris** [comment]
> this was discussed in the discord, but there shouldn't be alt titles for heads, it only adds for confusion

Little late for this comment don't you think? Make an issue so this doesn't get lost, kthx.

**Comment by SynnGraffkin** [comment]
I mean, I even brought that up but okay.

**Pull request merged**